fs.go

```diff
@@ -21,12 +21,16 @@ type fsWatcher struct {
 }
 
 func (ca *Cagent) FSWatcher() *fsWatcher {
-	fsWatcher := fsWatcher{AllowedTypes: map[string]struct{}{}, ExcludedPathCache: map[string]bool{}, cagent: ca}
+	if ca.fsWatcher != nil {
+		return ca.fsWatcher
+	}
+
+	ca.fsWatcher = &fsWatcher{AllowedTypes: map[string]struct{}{}, ExcludedPathCache: map[string]bool{}, cagent: ca}
```

Comment:
Please use multi-line version to initialise the struct

Reply:
fixed